articles/validating-xml-using-dtd/index.md

```diff
@@ -0,0 +1,250 @@
+
+```xml
+<!DOCTYPE customer SYSTEM "validation.dtd">
+```
+
+Having understood various components of DTD, let's start building a DTD validator.
+
+In DTD, the XML elements are declared as
+
+```xml
+<!ELEMENT element-name (element-content)>
+```
+
+For our example, the outermost tag `<customer />` holds all the input fields together. So, in DTD, we must specify the tag as `<customer />` containing fields with tags like `<firstname />`, `<lastname />`, `<companyname />`, `<email />` and `<message />`. According to the DTD syntax, we must specify that `<customer />` tag as the parent element holding other child tags.
+
+```xml
+<!ELEMENT customer (firstname, lastname, companyname, email, message)>
+```
+
+After specifying the tags present inside the outermost tag. Now, we have to define the type of value that can be parsed for each tag. Here, all the input fields can contain data as text. According to DTD syntax, we specify it as `#PCDATA`, under each element. Since it is required to parse for validating if the values are textual data or not.
+
+```xml
+<!ELEMENT firstname (#PCDATA)>
+<!ELEMENT lastname (#PCDATA)>
+<!ELEMENT companyname (#PCDATA)>
+<!ELEMENT email (#PCDATA)>
+<!ELEMENT message (#PCDATA)>
+```
+
+##### Complete code
+If it is an Internal DTD validation, the whole XML code with its DTD validation is shown below
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE customer [
+<!ELEMENT customer (firstname, lastname, companyname, email, message)>
+<!ELEMENT firstname (#PCDATA)>
+<!ELEMENT lastname (#PCDATA)>
+<!ELEMENT companyname (#PCDATA)>
+<!ELEMENT email (#PCDATA)>
+<!ELEMENT message (#PCDATA)>
+]>
+<customer>
+  <firstname>John</firstname>
+  <lastname>Doe</lastname>
+  <companyname>Section</companyname>
+  <email>johndoe@section.io</email>
+  <message>Welcome message</message>
+</customer>
+```
+
+If it is an External DTD validation, the DTD code can be stored in a file called `validation.dtd`
+
+```xml
+<!ELEMENT customer (firstname, lastname, companyname, email, message)>
+<!ELEMENT firstname (#PCDATA)>
+<!ELEMENT lastname (#PCDATA)>
+<!ELEMENT companyname (#PCDATA)>
+<!ELEMENT email (#PCDATA)>
+<!ELEMENT message (#PCDATA)>
+```
+
+And, now the DTD file can be referenced in XML as shown below
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE customer SYSTEM "validation.dtd">
+<customer>
+  <firstname>John</firstname>
+  <lastname>Doe</lastname>
+  <companyname>Section</companyname>
+  <email>johndoe@section.io</email>
+  <message>Welcome message</message>
+</customer>
+```
+
+In both types of validations, the parsing is done by DTD based on the structure that defines the XML file.
+
+### Conclusion
+We had an overview of what Data serialization is, how XML files are being built, and its validation using DTD. This article serves only as an introduction to the validation of XML using DTD. It is highly recommended to try out the code manually by reading further from the referenced articles.
```

Comment:
You could specify the reader briefly about how to use a tool to check if the XML file is valid or not (An online tool would is also fine)

Reply:
This has been specified in Line 70 already. Anyway, I will mention about it in conclusion too